pre_commit/languages/rust.py

```diff
@@ -52,15 +90,57 @@ def _add_dependencies(
         f.truncate()
 
 
+def install_rust_with_toolchain(toolchain: str) -> None:
+    # Check if rustup is already present in PATH
+    rustup_exe = parse_shebang.find_executable('rustup')
+    with tempfile.TemporaryDirectory() as rustup_dir:
+        rustup_env = (
+            ('RUSTUP_HOME', rustup_dir),
+        )
+        with envcontext(rustup_env):
+            if rustup_exe is None:
+                # We did not detect rustup and need to download it first.
+                if sys.platform == 'win32':  # pragma: no cover
+                    if platform.machine() == 'x86_64':
+                        url = 'https://win.rustup.rs/x86_64'
+                    else:
+                        url = 'https://win.rustup.rs/i686'
+                else:
+                    url = 'https://sh.rustup.rs'
+
+                rustup_init_exe = os.path.join(
+                    rustup_dir,
+                    win_exe('rustup-init'),
+                )
+                urllib.request.urlretrieve(url, rustup_init_exe)
+                make_executable(rustup_init_exe)
+                _, _, _ = cmd_output_b(
+                    rustup_init_exe, '-y', '--quiet', '--no-modify-path',
+                    '--default-toolchain', 'none',
+                    retcode=0,
+                )
+
+                # Now, rustup should be present in `$CARGO_HOME/bin`.
+                rustup_exe = parse_shebang.find_executable('rustup')
+
+            if rustup_exe is None:
+                raise AssertionError(
+                    'failed to find rustup even though it should have been '
+                    'installed at this point -- this is probably a bug',
+                )
+
+            _, _, _ = cmd_output_b(
```

Comment:
you don't need to assign the return value here

Reply:
ok, this is a bit embarassing :hand_over_mouth:

Reply:
https://github.com/pre-commit/pre-commit/pull/2534/commits/5250fbbf932e8312d7bb7b291221fe4e1f7fa648